contracts/HubPool.sol

```diff
@@ -278,24 +285,85 @@ contract HubPool is Testable, Lockable, MultiCaller, Ownable {
         emit RelayerRefundExecuted(relayerRefundRequestId, poolRebalance, msg.sender);
     }
 
-    function disputeRelayerRefund(uint256 relayerRefundRequestId) public {
-        RelayerRefundRequest storage relayerRefund = relayerRefundRequests[relayerRefundRequestId];
-        require(
-            getCurrentTime() > relayerRefundRequests[relayerRefundRequestId].requestExpirationTimestamp,
-            "Passed liveness"
+    function disputeRelayerRefund() public {
+        require(getCurrentTime() <= refundRequest.requestExpirationTimestamp, "Request passed liveness");
+
+        // Request price from OO and dispute it.
+        uint256 totalBond = _getBondTokenFinalFee() + bondAmount;
+        bondToken.safeTransferFrom(msg.sender, address(this), totalBond);
+        // This contract needs to approve totalBond*2 against the OO contract. (for the price request and dispute).
+        bondToken.safeApprove(address(_getOptimisticOracle()), totalBond * 2);
+        _getOptimisticOracle().requestAndProposePriceFor(
+            identifier,
+            uint32(getCurrentTime()),
+            _getRefundProposalAncillaryData(),
+            bondToken,
+            // Set reward to 0, since we'll settle proposer reward payouts directly from this contract after a relay
+            // proposal has passed the challenge period.
+            0,
+            // Set the Optimistic oracle proposer bond for the price request.
+            bondAmount,
+            // Set the Optimistic oracle liveness for the price request.
+            refundProposalLiveness,
+            refundRequest.proposer,
+            // Canonical value representing "True"; i.e. the proposed relay is valid.
+            int256(1e18)
         );
 
-        // Delete the last element in the relayerRefundRequests array. This acts to throw out the request.
-        emit RelayerRefundDisputed(relayerRefundRequestId, msg.sender);
+        // Dispute the request that we just sent.
+        SkinnyOptimisticOracleInterface.Request memory ooPriceRequest = SkinnyOptimisticOracleInterface.Request({
+            proposer: refundRequest.proposer,
+            disputer: address(0),
+            currency: bondToken,
+            settled: false,
+            proposedPrice: int256(1e18),
+            resolvedPrice: 0,
+            expirationTime: getCurrentTime() + refundProposalLiveness,
+            reward: 0,
+            finalFee: _getBondTokenFinalFee(),
+            bond: bondAmount,
+            customLiveness: refundProposalLiveness
+        });
+
+        _getOptimisticOracle().disputePriceFor(
+            identifier,
+            uint32(getCurrentTime()),
+            _getRefundProposalAncillaryData(),
+            ooPriceRequest,
+            msg.sender,
+            address(this)
+        );
 
-        relayerRefundRequests.pop();
+        emit RelayerRefundDisputed(msg.sender, ooPriceRequest, refundRequest);
```

Comment:
Can we enumerate these structs for better parseability? Also might be nice to do so so we can drop fields that are unnecessary.

Reply:
will do this on a follow up.